lib/utils/server/feature-flags.ts

```diff
@@ -0,0 +1,17 @@
+import { PostHog } from "posthog-node";
+
+export type FeatureFlag = "contributions_evolution_by_type";
+
+export async function getAllFeatureFlags(userId: number) {
+  const client = new PostHog(
```

Comment:
Based off of https://posthog.com/docs/libraries/next-js

Reply:
Should PostHog be instantiated on each function call?

Reply:
The example in their docs that in linked to send to indicate yes, e.g.

```

export async function getServerSideProps(ctx) {

  const session = await getServerSession(ctx.req, ctx.res)
  let flags = null

  if (session) {
    const client = new PostHog(
      process.env.NEXT_PUBLIC_POSTHOG_KEY,
      {
        host: process.env.NEXT_PUBLIC_POSTHOG_HOST,
      }
    )

    flags = await client.getAllFlags(session.user.email);
    client.capture(session.user.email, 'loaded blog article', { url: ctx.req.url })

    await client.shutdownAsync()
  }
  ...
}
```